### PR TITLE
US137395 Add Settings Page

### PR DESCRIPTION
**Pull request by Vitalii-Misechko**
* Functional Testing 
  * [x] Browsers (Chrome, FF, Edge)
  * [x] Accessibility
  * [x] RTL
  * [x] Responsive
  * [x] Test cases
    * [x] A user can navigate to the setting page
    * [x] A user can go to the main page by hitting browser's back button
    * [x] A user can go to the main page by hitting Back to Adoption Dashboard link on the navbar
* Security, devops, perf: N/A
* UX and docs: N/A

* [x] Demo

### Demo

settings-page2

### Updated view of 2l-overflow-group

image

### CODE REVIEW

File: pages/adoption-dashboard-view.js [hunk 1]
```diff
@@ -86,7 +87,7 @@ class AdoptionDashboardView extends SkeletonMixin(Localizer(MobxLitElement)) {
 				flex-grow: 1;
 				margin-right: -0.3rem;
 				margin-top: 0.7em;
-				max-width: 270px;
+				max-width: 400px;
```
Comment (johngwilkinson) [inline pages/adoption-dashboard-view.js:90]: I wonder if this fixes the issue @NicholasHallman found with long French translations

**Review by johngwilkinson — COMMENTED** [review]
We need to be careful not to ship this in MI2. Ideas:
- we could make an MI2 branch now and merge the remaining work there
- we could make an MI3 branch now and merge PRs like this there

File: demo/ad-demo-view.js [hunk 2]
```diff
@@ -33,6 +34,10 @@ class AdDemoView extends LitElement {
 		if (hashMap.has('dir')) {
 			document.documentElement.setAttribute('dir', hashMap.get('dir'));
 		}
+
+		if (hashMap.has('settings')) {
+			redirect('/settings');
```
[the remark this answers is not in the document]
Comment (NicholasHallman) [reply demo/ad-demo-view.js:39]: Changing the url can cause a hard reload, this function pushes a history state that the router can observe

File: pages/adoption-dashboard-view.js (hunk 1 above)
Comment (NicholasHallman) [reply pages/adoption-dashboard-view.js:90]: It sound partially yep, but it's introduced some alignment issues we'll need to resolve. Probably by overriding the overflow groups css and using flex-end alignment

**Comment by johngwilkinson** [comment]
Testing on draco2: looks good.

One minor thing that might be worth fixing if it's not too hard: if I go directly to the settings page (e.g. hit f5), it loads the server data. Since the settings might affect what server data loads (fetching only some resultSets), and just to save on calls, it would be nice to load the server data only when it's needed for the main page.

**Comment by Vitalii-Misechko** [comment]
> We need to be careful not to ship this in MI2. Ideas:
> 
> * we could make an MI2 branch now and merge the remaining work there
> * we could make an MI3 branch now and merge PRs like this there

I've just created m3-main branch and set target for this PR

File: demo/ad-demo-view.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (johngwilkinson) [reply demo/ad-demo-view.js:39]: But this is the demo page and the hashes are for testing: why not have the test use `/settings` instead of `#settings`?

**Comment by Vitalii-Misechko** [comment]
> Testing on draco2: looks good.
> 
> One minor thing that might be worth fixing if it's not too hard: if I go directly to the settings page (e.g. hit f5), it loads the server data. Since the settings might affect what server data loads (fetching only some resultSets), and just to save on calls, it would be nice to load the server data only when it's needed for the main page.

Fixed

**Review by johngwilkinson — APPROVED** [review]

**Pull request merged**